src/proto/colors.rs

```diff
@@ -0,0 +1,45 @@
+//! An extension trait that provides the ability to strip IRC colors from a string
+use regex::Regex;
+
+/// An extension trait giving strings a function to strip IRC colors
+pub trait FormattedString {
+
+    /// Returns true if the string contains color, bold, underline or italics
+    fn is_formatted(&self) -> bool;
+
+    /// Returns the string with all color, bold, underline and italics stripped
+    fn strip_colors(&self) -> String;
+
+    const STRIP_COLORS: Regex = Regex::new(r"\x1F|\x02|\x12|\x0F|\x16|\x03(?:\d{1,2}(?:,\d{1,2})?)?").unwrap();
+
+    const BOLD: char = '\x02';
+    const UNDERLINE: char  = '\x1F';
+    const REVERSED: char = '\x16';
+    const NORMAL: char = '\x0F';
+    const COLOR: char = '\x03';
+
+}
+
+
+impl<'a> FormattedString for &'a str {
+    fn is_formatted(&self) -> bool {
+        self.contains(BOLD) ||
+            self.contains(UNDERLINE) ||
+            self.contains(REVERSED) ||
+            self.contains(NORMAL) ||
+            self.contains(COLOR)
+    }
+
+    fn strip_colors(&self) -> str {
```

Comment:
My brain type checker could be failing me, but I think the return type here is actually `Cow<str>`. So, you should be able to `use std::borrow::Cow` and change the return type in the impls and the trait definition to `Cow<str>`.